### DISCUSSION TITLE
A way to validate hidden input with client side

### DISCUSSION THREAD

**Discussion by Forus-Spec** [edmundhung/conform#364] [Ideas]
Hi Conform Team,

I've been using Conform and find it amazing! I have a suggestion for a feature that could enhance its functionality. Currently, Conform does a great job of validating visible inputs, but I was wondering if there could be a possibility of extending this validation to hidden inputs as well.

Here's the scenario I encountered: I have a use case where I fetch data from a combobox, and upon selecting an item, I default hidden inputs to specific values. Essentially, I have a Fieldset that holds two properties: 

If the user selects something, these values are automatically populated:

__ simplification for a mental model __

```javascript
    {...conform.input(nestedField.name)}
    {...conform.input(nestedField.id,{type:"hidden")} 
```
If the user select an item through combobox given options:
```javascript
{
  itemName: "exampleName",    
  itemId: "exampleId"  
} (client side validated ✔️ | server side validated ✔️)
``` 
If the user doesn't select anything and just enters random values, it might look like this:
```javascript
{
  itemName: "blablabla",  // this will be client-validated
  itemId: ""  // this won't be validated since it's hidden and won't default to anything since nothing was naturally selected
} (client side validated ❌  | server side validated  ✔️)
```
The issue arises because the item name input can be anything, and it didn't get picked from valid options 
I understand that there might be various solutions to this problem 
but I prefer this one as it aligns with the principles of progressive enhancement.

Adding a feature to Conform or pathway that allows for the validation of hidden inputs in client side 
would greatly enhance its flexibility and usefulness in scenarios like the one I described.

 I appreciate your ongoing efforts Conform.

**Comment by edmundhung** [comment]
Hi @Forus-Spec, this will be resolved in the next version. The issue you see is caused by the usage of `setCustomValidity()` in Conform to propagate errors. However this web API does not work consistently on all input type and, in your case, the hidden input is ignored by the browser.

We are migrating to a context approach going forward. For details you can see this [RFC](https://github.com/edmundhung/conform/discussions/331) and also test out the hidden input support using the pre-release version.

**Reply by Forus-Spec** [reply]
Amazing and awesome features !
